tests/pagination.py

```diff
@@ -75,3 +75,30 @@ def test_paginate_function(app):
         def foo_instance():
             return Cursor([1, 2, 3])
         assert paginate(foo_instance, skip=2)().data == [3]
+
+
+@pytest.mark.parametrize("app", [
+    Stupeflask(__name__),
+    Flask(__name__)
+])
+def test_paginate_header_total_count(app):
+    with app.test_request_context():
+        paginate(Cursor([1, 2, 3]), headers=True)
+        if isinstance(app, Stupeflask):
+            assert request.response_headers["X-Total-Count"] == 3
+            assert "Link" not in request.response_headers
+
+
+@pytest.mark.parametrize("app", [
+    Stupeflask(__name__),
+    Flask(__name__)
+])
+def test_paginate_header_link(app):
+    with app.test_request_context():
+        paginate(Cursor([1, 2, 3]), limit=1, skip=0, headers=True)
+        if isinstance(app, Stupeflask):
+            links = request.response_headers["Link"].split(",")
+            assert links[0].split("?")[1] == 'limit=1&skip=0>; rel="first"'
```

Comment:
Honestly it's good enough for me. If you prefer, an alternative could be something like this:

```suggestion
            assert "limit=1" in links[0]
            assert "skip=0" in links[0]
```